packages/go_router/lib/src/router.dart

```diff
@@ -526,10 +526,16 @@ class GoRouter implements RouterConfig<RouteMatchList> {
       // verified by assert() during the initialization.
       return initialLocation!;
     }
-    final Uri platformDefaultUri =
-        Uri.parse(WidgetsBinding.instance.platformDispatcher.defaultRouteName);
-    final String platformDefault =
-        platformDefaultUri.path.isEmpty ? '/' : platformDefaultUri.path;
+    Uri platformDefaultUri = Uri.parse(
+      WidgetsBinding.instance.platformDispatcher.defaultRouteName,
+    );
+    if (platformDefaultUri.hasEmptyPath) {
+      platformDefaultUri = Uri(
```

Comment:
this can be `platformDefaultUri.replace(path: '/')`

Reply:
That doesn't do the same.

For example, if `platformDefaultUri` is `https://domain.com?param=1`, my current implementation creates the Uri `/?param=1`, while using `Uril.replace(path: '/')` creates `https://domain.com/?param=1`. This breaks a handful of tests, and I think go_router expects `/?param=1` rather than `https://domain.com/?param=1`?

Maybe I could change those lines into:

```dart
    Uri platformDefaultUri = Uri.parse(
      WidgetsBinding.instance.platformDispatcher.defaultRouteName,
    );
    platformDefaultUri = Uri(
      path: platformDefaultUri.hasEmptyPath ? '/' platformDefaultUri.path,
      queryParameters: platformDefaultUri.queryParameters,
    );
    final String platformDefault = platformDefaultUri.toString();
```

What do you think ?

Reply:
this will be feed into RouteInformation.location so I think you are right, we shouldn't include the scheme and host. Can you add a todo here to remind us that this can be clean up once RouteInformation.uri is available in packages repo?

Reply:
I added the todo in [doc: Add todo](https://github.com/flutter/packages/pull/5276/commits/9fa3c3778dfaa68a039ab8f670aba9b9733e7e51)
I put your username. Tell if I should put another one